apps/roam/src/components/DiscourseNodeMenu.tsx

```diff
@@ -42,15 +41,15 @@ const NodeMenu = ({
   extensionAPI,
   trigger,
   isShift,
-  isTextSelected,
 }: { onClose: () => void } & Props) => {
-  const [showNodeTypes, setShowNodeTypes] = useState(isTextSelected || isShift);
+  const [showNodeTypes, setShowNodeTypes] = useState(!!isShift);
+  const userDiscourseNodes = useMemo(
+    () => getDiscourseNodes().filter((n) => n.backedBy === "user"),
+    [],
+  );
   const discourseNodes = useMemo(
-    () =>
-      getDiscourseNodes().filter(
-        (n) => n.backedBy === "user" && (showNodeTypes || n.tag),
-      ),
-    [showNodeTypes],
+    () => userDiscourseNodes.filter((n) => showNodeTypes || n.tag),
```

Comment:
Because `getDiscourseNodes` is memoized (which is the expensive query), `discourseNodes` doesn't also need to be memoized because it is a simple filter.